core/modules/dblog/dblog.install

```diff
@@ -107,3 +107,33 @@ function dblog_update_1001() {
   $short = system_date_format_load('short');
   config_set('system.core', 'log_date_format', $short['pattern']);
 }
+
+/**
+ * Convert the format for dates in the database log.
+ */
+function dblog_update_1002() {
+  $t = get_t();
+  $config = config('system.core');
+  $log_format = $config->get('log_date_format');
+
+  // Look for a matching date format to use.
+  foreach (system_get_date_formats() as $date_format) {
+    if ($log_format == $date_format['pattern']) {
+      $config->set('log_date_format', $date_format['name']);
+      $config->save();
+      break;
+    }
+  }
+
+  // No matching date format found, so create a custom one.
+  if ($config->get('log_date_format') == $log_format) {
```

Comment:
Not sure this conditional is required 🤔 ...we're setting `$log_format = $config->get('log_date_format');` earlier, and we also `break;` from the `foreach` if `$log_format` matches an existing pattern. I think that there's no point - we should directly do the `system_date_format_save()`.

Reply:
This is needed. We first look for a matching date format and, if found, use it. Then we're done, nothing more needed. If, after going through all existing date formats, a match isn't found (i.e. log_date_format in config still equals the old pattern value) then we need to create and save a new date format, then save its name in config.

Does that explain better?

Reply:
Yeah, sorry, I confused `break` with `return` here. Ignore me.

Reply:
Ah yes, a `return;` would simplify things, but when I looked into the `hook_update_N()` function, it said returns will be printed to the user as a message, so I thought it better not to do a return then.